src/com/google/common/css/compiler/ast/GssParserCC.jj

```diff
@@ -707,6 +707,8 @@ PARSER_END(GssParserCC)
   // Special handling needed because ':not' takes simple selectors as
```

Comment:
Aye.  I agree that the duplication here is less than stellar.

However, I think the semantics of all three are slightly different.
:not() takes only a simple selector; one tag name, class name or id.
::slotted() takes [only a compound selector](https://github.com/w3c/webcomponents/issues/331#issuecomment-169200030); tag.class would be valid (can select only a single element though)
:host() takes any full selector (allows combinators).